src/drafts/MarkdownEditor/_useListEditing.ts

```diff
@@ -45,22 +46,23 @@ const isNumericListItem = (item: ListItem | null): item is NumericListItem => ty
 export const parseListItem = (line: string): ListItem | null => {
   const result = listItemRegex.exec(line)
   if (!result) return null
-  const [, leadingWhitespace = '', fullDelimeter, itemNumberStr = '', taskBox = null, text] = result
+  const [, leadingWhitespace = '', fullDelimeter, itemNumberStr = '', middleWhitespace, taskBox = null, text] = result
   const itemNumber = Number.parseInt(itemNumberStr, 10)
   const delimeter = Number.isNaN(itemNumber) ? (fullDelimeter as '*' | '-') : itemNumber
 
   return {
     leadingWhitespace,
     text,
     delimeter,
+    middleWhitespace,
     taskBox: taskBox as '[ ]' | '[x]' | null,
   }
 }
 
 export const listItemToString = (item: ListItem) =>
   `${item.leadingWhitespace}${typeof item.delimeter === 'number' ? `${item.delimeter}.` : item.delimeter}${
-    item.taskBox ? ` ${item.taskBox}` : ''
-  } ${item.text}`
+    item.middleWhitespace
+  }${item.taskBox ? ` ${item.taskBox}` : ''} ${item.text}`
```

Comment:
Since we are preserving the `middleWhitespace` now we don't need this additional space:

```suggestion
  }${item.taskBox || ''} ${item.text}`
```

Reply:
ah good catch